frontend/src/lib/search-utils.ts

```diff
@@ -22,32 +22,61 @@ export const handleStandardComparison = (valueOne: string, valueTwo: string, ope
   }
 }
 
-// for a given displayVersion string there is a distribution value a ' ' and a semver value, here we divide them and take the semver
-export const handleSemverOperatorComparison = (versionOne: string, versionTwo: string, operator: SearchOperator) => {
-  // Semver coerces the version to a valid semver version if possible, otherwise it returns the original value
-  const coercedVersionOne = semver.valid(semver.coerce(versionOne)) ?? versionOne
-  const coercedVersionTwo = semver.valid(semver.coerce(versionTwo)) ?? versionTwo
+export const handleSemverOperatorComparison = (
+  versionToCompare: string,
+  inputVersion: string,
+  operator: SearchOperator
+) => {
+  const coerceVersion = (version: string) => semver.valid(semver.coerce(version)) ?? version
+  const coercedInputVersion = coerceVersion(inputVersion)
+  const coercedVersionToCompare = coerceVersion(versionToCompare)
 
-  const validInputSemvers = !!semver.valid(coercedVersionOne) && !!semver.valid(coercedVersionTwo)
-  if (!validInputSemvers) {
-    if (operator === SearchOperator.NotEquals) {
-      return true
+  // If the inputVersion is not a valid semver, but the coerced version is, we can use the incomplete version to create a range
+  if (!semver.valid(inputVersion) && semver.valid(semver.coerce(inputVersion))) {
+    const isMajorVersionAlone = !inputVersion.includes(
+      `${semver.coerce(inputVersion)?.major}.${semver.coerce(inputVersion)?.minor}`
+    )
+    const topRange = isMajorVersionAlone
+      ? semver.inc(coercedInputVersion, 'major')
+      : semver.inc(coercedInputVersion, 'minor')
+
+    const equalsValueRange = `${coercedInputVersion} - ${topRange}`
+
+    switch (operator) {
+      case SearchOperator.Equals:
+        return semver.satisfies(coercedVersionToCompare, equalsValueRange)
+      case SearchOperator.GreaterThan:
+        return semver.satisfies(coercedVersionToCompare, `>${coercedInputVersion}`)
+      case SearchOperator.GreaterThanOrEqualTo:
+        return semver.satisfies(coercedVersionToCompare, `>=${coercedInputVersion}`)
+      case SearchOperator.LessThan:
+        return semver.satisfies(coercedVersionToCompare, `<${coercedInputVersion}`)
+      case SearchOperator.LessThanOrEqualTo:
+        return semver.satisfies(coercedVersionToCompare, `<=${coercedInputVersion}`)
```

Comment:
Is there a reason we are changing these ones to use `satisfies` if they are not using a range?

Also, I think we should refactor this code to avoid repeating the switch statement.

Reply:
If I'm understanding you correctly, I believe using `satisfies` is still necessary in this situation, as this language (e.g. `>${coercedInputVersion}`)  is actually part of the [range grammar](https://github.com/npm/node-semver?tab=readme-ov-file#range-grammar), and would not work with something like `semver.gt`.

Otherwise, big copy-that on the refactor.

Reply:
But `coercedInputVersion` itself is not a range. It's only a range because we added the operator before it. I think it's a valid question of whether we want to use a range in this case or not though.

To test this, in your file I added `window.semver = semver` so I could easily call semver functions in the browser console.

Suppose a user has entered "version>4.12"

semver.gt("4.12.1", "4.12.0") is true
semver.satisfies("4.12.1" ">4.12") is false

To me the latter makes sense because the user did not specify the patch version, so they are looking for 4.13 or higher.

I opened a PR against your PR with simplified implementation.
https://github.com/Randy424/console/pull/20

Reply:
Also, this matches how npm does version ranges. See https://semver.npmjs.com/

Enter "lodash" and ">1" and you'll see the lowest version is 2.0.0.